broker/tests/broker_server_test.cpp

```diff
@@ -0,0 +1,421 @@
+/* SPDX-License-Identifier: BSD-2-Clause-Patent
+ *
+ * SPDX-FileCopyrightText: 2016-2020 the prplMesh contributors (see AUTHORS.md)
+ *
+ * This code is subject to the terms of the BSD+Patent license.
+ * See LICENSE file for more details.
+ */
+
+#include <gtest/gtest.h>
+
+#include "../src/broker_server.h"
+
+#include <bcl/beerocks_os_utils.h>
+
+#include <beerocks/tlvf/beerocks_message.h>
+#include <beerocks/tlvf/beerocks_message_broker.h>
+
+#include <tlvf/CmduMessageTx.h>
+
+std::string colorize(std::string msg);
+#define ELPP_CUSTOM_COUT_LINE(msg) colorize(msg)
+#include <easylogging++.h>
+
+// Initialize easylogging++
+INITIALIZE_EASYLOGGINGPP
+
+namespace beerocks {
+namespace broker {
+namespace tests {
+
+//////////////////////////////////////////////////////////////////////////////
+////////////////////////////// Global Variables //////////////////////////////
+//////////////////////////////////////////////////////////////////////////////
+
+// UDS file for the tests
+constexpr char broker_uds_file[] = "beerocks_broker_test_uds";
+
+// Buffer for creating and sending messages
+static uint8_t s_tx_buffer[message::MESSAGE_BUFFER_LENGTH];
+
+//////////////////////////////////////////////////////////////////////////////
+/////////////////////////////// Helper Classes ///////////////////////////////
+//////////////////////////////////////////////////////////////////////////////
+
+// Wrapper class for the broker_server_thread for capturing errors
+class broker_server_thread_wrapper : public broker_server_thread {
+public:
+    broker_server_thread_wrapper(std::string broker_uds_path)
+        : broker_server_thread(broker_uds_path)
+    {
+    }
+
+    bool error() { return error_occurred; }
+
+protected:
+    bool error_occurred = false;
+
+    virtual bool handle_msg(std::shared_ptr<Socket> sd) override
+    {
+        if (broker_server_thread::handle_msg(sd) == false) {
+            error_occurred = true;
+            return false;
+        }
+
+        error_occurred = false;
+        return true;
+    }
+};
+
+//////////////////////////////////////////////////////////////////////////////
+/////////////////////////////////// Tests ////////////////////////////////////
+//////////////////////////////////////////////////////////////////////////////
+
+TEST(broker_server, invalid_message_magic)
+{
+    broker_server_thread_wrapper broker_server(broker_uds_file);
+    ASSERT_TRUE(broker_server.start("broker"));
+
+    // Create an empty CMDU
+    ((message::sUdsHeader *)(s_tx_buffer))->magic = 12345678;
+    ieee1905_1::CmduMessageTx cmdu(s_tx_buffer + sizeof(message::sUdsHeader), sizeof(s_tx_buffer));
+
+    // Build a subscribe message
+    auto subscribe =
+        message_com::create_vs_message<beerocks_message::cACTION_BROKER_SUBSCRIBE>(cmdu);
+
+    ASSERT_TRUE(subscribe != nullptr);
```

Comment:
Don't we need to unsubscribe as part of the test?
Have you tried to run the tests with Valgrind? This is something yet to be added to CI pipeline too (see #1348)

Reply:
Not in this test. It's purpose is to only test handling on invalid magic values